### DISCUSSION TITLE
cannot find -lvtksys-9.1

### DISCUSSION THREAD

**Discussion by lizhen-dlut** [idaholab/moose#24782] [Q&A Getting Started] [answered]
## Bug Description

when i use command 
cd ~/projects/moose/test
make -j 6

Linking Library /mnt/d/projects/moose/framework/libmoose-opt.la...
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkIOCore-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkCommonCore-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkCommonDataModel-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkFiltersCore-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkIOXML-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkImagingCore-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkIOImage-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkImagingMath-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkIOParallelXML-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkParallelMPI-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkParallelCore-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtkCommonExecutionModel-9.1: No such file or directory
/home/lizhen/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lvtksys-9.1: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [/mnt/d/projects/moose/framework/moose.mk:390: /mnt/d/projects/moose/framework/libmoose-opt.la] Error 1
make: *** Waiting for unfinished jobs....

**Comment by milljm** [comment, marked as answer]
Converting to a Discussion. An "issue" which would indicate something that needs to change in the MOOSE repository in order to fix something that is plaguing everyone.

This happens due to your Conda environment being slightly out of date (we've updated the `moose-libmesh-vtk` to be 9.2.6).

While I would love to solve what exactly is causing the issue, historically the solution has been to re-create the moose environment. You can of course try to perform an update, but I suspect you've already tried that!:

```bash
mamba deactivate
mamba activate base
mamba env remove -n moose
mamba create -n moose moose-dev
```

Once re-created, it is also wise to clean your MOOSE repo, perform an update, and attempt a build again:

```bash
cd /mnt/d/projects/moose
git clean -xfd
git pull
cd test
make -j 6
```

**Reply by lizhen-dlut** [reply]
Thanks, the compilation problem is solved. Yesterday I thought it was a system problem and I reinstalled Ubuntu.
